### PR TITLE
Fixing type ambiguity of `unbroadcast`

### PR DESCRIPTION
**Pull request by SebastianAment**
Fixing #1184.

### CODE REVIEW

**Comment by mcabbott** [comment]
Looks good.

Is it easy to trigger this problem? Would be nice to have a test, ideally calling `gradient` of something.

**Comment by SebastianAment** [comment]
Great, I just came up with this:
```Julia
using Zygote
n, d = 128, 3
x = [randn(d) for _ in 1:n]

f = sin
g(x) = sum.((f,), x)
h(x) = sum(abs2, g(x))
gradient(h, x)
```
Which triggers the error. Will add it to the testing suite.

File: test/gradcheck.jl [hunk 1]
```diff
@@ -1487,17 +1487,6 @@ using Zygote: Buffer
   @test ∇x == 6 .* x
 end
 
-@testset "FillArrays" begin
```
Comment (SebastianAment) [inline test/gradcheck.jl:1490]: There were two separate "FillArrays" test sets, I moved the code here without modifications to the second test set starting line 1614.

**Comment by SebastianAment** [comment]
I think this is ready to be merged.

**Review by ToucheSir — APPROVED** [review]
Thanks!

**Pull request merged**